### PR TITLE
Orand - Allows randomness to be generated and fed to any smart contracts on any EVM compatible blockchain.

### PR DESCRIPTION
**Pull request by hubertnguyen88**
# Request for Proposals

## Abstract

Orand is an Elliptic Curve Verifiable Random Function (ECVRF: https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-vrf-05) as a service that offers high performance, zero latency and 30 times lower cost than current providers. Orand offers on-chain verification and pseudorandom for projects that seek integrity and transparency in their features with a wide variety of case studies from card games, MMORPG to NFT collectibles
## Background

Protect the project from manipulation and hack that can change the outcome of the results
Provide trustless randomness to increase prestige among the community

## Checklist

- [X ] I have checked the [RFPs](https://github.com/w3f/Grants-Program/tree/master/docs/rfps) to make sure this is not a duplicate.
- [ X] I have read and followed the [RFP Suggestion instructions](https://github.com/w3f/Grants-Program#mailbox_with_mail-suggest-a-project).

### CODE REVIEW

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for creating the RFP. But I’m not fully sure how this is beneficial for the polkadot and kusama ecosystem. Could you further clarify this as part of the RFP? Also, you mentioned pallets as part of the deliveries, but why do you focus only on evm parachains? Additionally, are you personally in contact with the Orochi Network team? Are they interested in bringing their technology to the polkadot ecosystem?  What is the benefit of this solution compared to others?

**Comment by hubertnguyen88** [comment]
> Thanks for creating the RFP. But I’m not fully sure how this is beneficial for the polkadot and kusama ecosystem. Could you further clarify this as part of the RFP? Also, you mentioned pallets as part of the deliveries, but why do you focus only on evm parachains? Additionally, are you personally in contact with the Orochi Network team? Are they interested in bringing their technology to the polkadot ecosystem? What is the benefit of this solution compared to others?

Hi @Noc2 , thank you for your feedback.
I would like to response each questions as below:

1. But I’m not fully sure how this is beneficial for the polkadot and kusama ecosystem. Could you further clarify this as part of the RFP?
-This transformation of Web3 Gaming requires that the game's computational processes become verifiable. Every mechanism of the game is built around random numbers, such as rare NFT drop rate, damage ability, item stats...all need randomness. The Verifiable Random Function is therefore an essential component of Web3 Gaming to make Web3 Gaming fairer. https://www.wsgr.com/en/insights/recent-rulings-suggest-defendant-wins-in-loot-box-cases-are-common-appeals-all-pending.html
-Building an ECVRF powered pallet can help Polkadot provide verifiable randomness for most parachains as well as entire ecosystems on parachains.
-MPC ECVRF can be used to secure Polkadot consensus leader selection process.

2. Also, you mentioned pallets as part of the deliveries, but why do you focus only on evm parachains?
Our side provides 2 pallets, 1 pallet for ECVRF in general and 1 pallet for EVM compatible chain. Supporting EVM parachains because there are many technical advantages such as on-chain verfication as well as a large number of dApps. If desired, we can also support Wasm parachains, but need to adjust the total cost of the project. Because we need to implement one more verifier for Wasm parachain.

3. Additionally, are you personally in contact with the Orochi Network team?
I'm Hubert Nguyen - Chief Growth Officer of Orochi Network, you can check github of our founder here https://github.com/chiro-hiro

4. What is the benefit of this solution compared to others?
There aren't many MPC EVRF or Decentralized RNG projects on the market right now. Drand is a MPC VRF development project based on Boneh-Lynn-Shacham (BLS) signatures, but on-chain verifiers are very expensive on Polkadot parachains.

I hope above responses are clear, if you need more information, just let me know.

File: docs/RFPs/Open/orand.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+**Fault Proof**: If the game/application server tries to delay the feeding process to manipulate the result, a fault proof will be committed so sue the game server.
+
+**Multi-chain**: All EVM compatible blockchains can be supported.
+
+## Deliverables :nut_and_bolt:
+
+Please list the deliverables of the project in as much detail as possible. Please also estimate the amount of work required and try to divide the project into meaningful milestones.
+
+* **Total Estimated Duration:** 19
```
Comment (semuelle) [inline docs/RFPs/Open/orand.md:27]:
```suggestion
* **Total Estimated Duration:** 19 months
```

File: docs/RFPs/Open/orand.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+**Fault Proof**: If the game/application server tries to delay the feeding process to manipulate the result, a fault proof will be committed so sue the game server.
+
+**Multi-chain**: All EVM compatible blockchains can be supported.
+
+## Deliverables :nut_and_bolt:
+
+Please list the deliverables of the project in as much detail as possible. Please also estimate the amount of work required and try to divide the project into meaningful milestones.
+
+* **Total Estimated Duration:** 19 
+* **Full-time equivalent (FTE):**  1,500
```
Comment (semuelle) [inline docs/RFPs/Open/orand.md:28]: I don't think there will be 1500 people working on this for 19 months. Are these 1500 person-days? Over 19 months, that would be about 2.6 FTEs.

**Review by semuelle — CHANGES_REQUESTED** [review]
Thanks for the PR, @hubertnguyen88. I have moved the file to the RFPs folder. Sorry for meddling with your fork, but I tried to submit a pull request and that didn't work.

I'm a bit confused, because you used the template for RFPs, but in your comments and in the doc you talk about what Orochi "can provide" and "adjusting costs". Are you planning to apply for this yourself? Because in that case, you don't have to submit an RFP, you can go straight to applying. You would have to use the [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) and the [application PR template](https://github.com/w3f/Grants-Program/blob/master/.github/pull_request_template.md), though.

File: docs/RFPs/Open/orand.md (hunk 1 above)
Comment (hubertnguyen88) [reply docs/RFPs/Open/orand.md:27]: Hi @semuelle , i thought we have to submit RFP first, then apply for grant later. So should i use application template and PR template which you shared?

Comment (semuelle) [reply docs/RFPs/Open/orand.md:27]: Hey @hubertnguyen88. Sorry for the late reply. RFPs are just for when you are looking for others to implement an idea. If you want to implement it yourself, then yes, please use the application template. The process and documents are detailed [here](https://w3f.github.io/Grants-Program/docs/Process/how-to-apply).

**Comment by Noc2** [comment]
Let me know if I should reopen the PR.

**Pull request closed without merging**